### PR TITLE
Custom mesh example

### PR DESCRIPTION
**Pull request by jcdickinson**
A currently failing attempt at manually allocating the vertex and index buffers. Looking for guidance so that this example can be committed for others.

### CODE REVIEW

File: examples/3d/custom_render.rs [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        },
+        renderer::{
+            BufferId, BufferInfo, BufferUsage, RenderContext, RenderResourceBindings,
+            RenderResourceContext, RenderResourceId, RenderResources,
+        },
+        shader::{ShaderStage, ShaderStages},
+    },
+};
+
+/// This example illustrates how to create a custom material and mesh asset and a shader that uses that uses those assets
```
Comment (Abendstolz) [inline examples/3d/custom_render.rs:19]: *nitpick*

> that uses that uses

repetition

File: examples/3d/custom_render.rs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                Vec3::new(0.0, 1.0, 0.0),
+            )),
+            ..Default::default()
+        });
+}
+
+pub fn my_mesh_node_system(
+    mut state: Local<MyMeshNodeState>,
+    render_resource_context: Res<Box<dyn RenderResourceContext>>,
+    mut render_resource_bindings: ResMut<RenderResourceBindings>,
```
Comment (Abendstolz) [inline examples/3d/custom_render.rs:222]: One thing that strikes immediately is that this is unused.

**Comment by multun** [comment]
Can you rebase this on master? Format checking was enabled, and I believe the CI doesn't yet check for it at the commit you're at
I realize this PR isn't yet ready, it's just for us not to forget when it is

File: examples/3d/custom_render.rs (hunk 2 above)
Comment (jcdickinson) [reply examples/3d/custom_render.rs:222]: This example is incomplete, I'll be doing lots of cleanup once it get it working.

**Comment by multun** [comment]
Awesome, thanks! (the failure isn't yours and was just fixed on master)

**Comment by jcdickinson** [comment]
Ha! That failure and quick fix really confused me :)

File: examples/3d/custom_render.rs [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+                buffer_usage: vertex.buffer_usage | BufferUsage::COPY_DST,
+                mapped_at_creation: false,
+                ..vertex
+            }),
+            index: render_resource_context.create_buffer(BufferInfo {
+                buffer_usage: index.buffer_usage | BufferUsage::COPY_DST,
+                mapped_at_creation: false,
+                ..index
+            }),
+            vertex_staging: render_resource_context.create_buffer(BufferInfo {
```
Comment (StarArawn) [inline examples/3d/custom_render.rs:90]: Just as a unrelated note, we likely should be using `wgpu::queue.write_buffer` for some stuff as it removes the need for a staging buffer.

Comment (jcdickinson) [reply examples/3d/custom_render.rs:90]: That's really helpful, thanks.

**Comment by joshuajbouw** [comment]
If it helps, I have made a custom mesh here https://github.com/joshuajbouw/bevy_tilemap/blob/master/src/mesh.rs which is a grid of vertices.

**Comment by jcdickinson** [comment]
I sadly don't have free time to chase this down right now :(.

**Pull request closed without merging**